extensions/amp-selector/1.0/README.md

```diff
@@ -0,0 +1,287 @@
+
+#### Actions
+
+The `bento-selector` API allows you to perform the following actions:
+
+**selectBy(delta: number)**
+Closes the selector.
+
+```js
+api.selectBy(1); // Select next option in DOM sequence.
+api.selectBy(-2); // Select the option that is two previous in DOM sequence.
+```
+
+**toggle(optionValue: string, opt_select: boolean|undefined)**
+Toggles the option with the given `optionValue` to be selected or deselected based on `opt_select`. If `opt_select` is not present, then the option will be selected if currently not selected, and deselected if currently selected.
+
+```js
+api.toggle("a"); // Toggle the item with the attribute `option="a"`.
+api.toggle("1", true); // Select the item with the attribute `option="1"`.
+```
+
+#### Events
+
+The `bento-selector` API allows you to register and respond to the following events:
+
+**select**
+
+This event is triggered when the user selects an option.
+Multi-selectors and single-selectors fire this when selecting or unselecting options.
+Tapping disabled options does not trigger the `select` event.
+
+<ul>
+  <li>
+  `event.data.targetOption` contains the `option` attribute value of the selected element.</li>
+  <li>
+  `event.data.selectedOptions` contains an array of the `option` attribute values of all selected elements.
+  </li>
+</ul>
+
+```js
+selector.addEventListener("select", (e) => console.log(e.data.targetOption))
+```
+
+### Preact/React Component
+
+The examples below demonstrate use of the `<BentoSelector>` as a functional component usable with the Preact or React libraries.
+
+#### Example: Import via npm
+
+[example preview="top-frame" playground="false"]
+
+Install via npm:
+
+```sh
+npm install @ampproject/bento-selector
+```
+
+```javascript
+import React from 'react';
+import { BentoSelector, BentoSelectorOption} from '@ampproject/bento-selector/react';
+import '@ampproject/bento-selector/styles.css';
+
+function App() {
+  return (
+    <BentoSelector aria-label="Option menu">
+      <BentoSelectorOption option="1">Option 1</BentoSelectorOption>
+      <BentoSelectorOption option="2">Option 2</BentoSelectorOption>
+      <BentoSelectorOption option="3">Option 3</BentoSelectorOption>
+      <BentoSelectorOption option="4">Option 4</BentoSelectorOption>
+    </BentoSelector>
+  );
+}
+```
+
+[/example]
+
+#### Layout and Style
+
+Each Bento component has a small CSS library you must include to guarantee proper loading without [content shifts](https://web.dev/cls/). Because of order-based specificity, you must manually ensure that stylesheets are included before any custom styles.
+
+```html
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/amp-facebook-1.0.css">
+```
+
+Alternatively, you may also make the light-weight pre-upgrade styles available inline:
+
+```html
+<style data-bento-boilerplate>
+ BentoSelector{
+    display: block;
+    overflow: hidden;
+    position: relative;
+  }
+</style>
+```
```

Comment:
The Preact section is not concerned with upgrading styles since it is not a web component.

Please instead use the section from other README's Preact section, which discusses how to apply style via `styles` or `className` props.